src/helpers/eventInstance.ts

```diff
@@ -0,0 +1,69 @@
+import type mongoose from "mongoose";
+import type {
+  InterfaceEvent,
+  InterfaceOrganization,
+  InterfaceUser,
+} from "../models";
+import { Event } from "../models";
+import type { MutationCreateEventArgs } from "../types/generatedGraphQLTypes";
+
+//Function for creating single events
+
+export async function generateSingleEvent(
+  args: Partial<MutationCreateEventArgs>,
+  currentUser: InterfaceUser,
+  organization: InterfaceOrganization,
+  session: mongoose.ClientSession
+): Promise<Promise<InterfaceEvent[]>> {
+  const createdEvent = await Event.create(
+    [
+      {
+        ...args.data,
+        creatorId: currentUser._id,
+        admins: [currentUser._id],
+        organization: organization._id,
+      },
+    ],
+    { session }
+  );
+
+  return createdEvent;
+}
+
+//Function for creating Weekly events
+
+export async function generateWeeklyEvents(
+  args: Partial<MutationCreateEventArgs>,
+  currentUser: InterfaceUser,
+  organization: InterfaceOrganization,
+  session: mongoose.ClientSession
+): Promise<InterfaceEvent[]> {
+  const recurringEvents = [];
+  const { data } = args;
+
+  const startDate = new Date(data?.startDate);
+  const endDate = new Date(data?.endDate);
```

Comment:
Where is the logic for no end date? It was supposed to January 1, 2100

Reply:
I have not added it because I believe it is not required. By default in our model, if `startDate` is given, `endDate` should also be required. I consider this a good practice.

Since we are following Google Calendar, where the end date is also required, the same principle applies to Talawa-admin/mobile. We do not provide users with an option to not give the endDate.

Consider creating `Daily` recurring events with no end date ; it will generate approximately 27,000 events.